lib/authHandlers.js

```diff
@@ -149,6 +151,25 @@ var sessions       = require('client-sessions'),
     }
   }
 
+  var handleSSOSetup = function(req, res, next) {
+    console.log("EREZ handleSSOSetup: ", req.query.access_token);
+    secondarySessionTokens[req.query.user_token] = req.query.access_token;
+    next();
+  }
+
+  var handleSSOLogin = function(req, res, next) {
+    console.log("EREZ handleSSOLogin: req.query = ", req.query);
+    if (req.query.user_token) {
+      console.log("EREZ handleSSOLogin: req has user token!!");
+      req.query.access_token = secondarySessionTokens[req.query.user_token];
```

Comment:
> What are the implications of not having a session token in the cache?
Do you mean what happens if `req.query.user_token` is not in `secondarySessionTokens`?
Then `req.query.access_token` will be set a `undefined` and the authentication using it will fail and the user will be redirected to authenticate with Openshift.